### PR TITLE
Simplify Core

### PR DESCRIPTION
**Pull request by f2prateek**
Worked on this a while back, feel free to close this if you already have something in the works. Ref #172.

### CODE REVIEW

**Comment by f2prateek** [comment]
Realized I had reformatted the structure, let me know if you want me to change it back.

**Comment by JakeWharton** [comment]
As painful as it is, we probably need to keep the four    space    indent    to    be    consistent.

Also per the linked issue we need to do a final deprecate-all-the-things release before we'd merge.

**Comment by f2prateek** [comment]
Rebased and reverted the formatting to the original. I tried to only delete things as much as I could, and not fix other issues I noticed (like formatting https://cloudup.com/cmAg0ka9yxE).

File: sample-app/src/main/java/rx/android/samples/MainActivity.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                    // Do some long running operation
+                    Thread.sleep(TimeUnit.SECONDS.toMillis(5));
+                } catch (InterruptedException e) {
+                    throw OnErrorThrowable.from(e);
+                }
+                return Observable.just("one", "two", "three", "four", "five");
+            }
+        }) //
+                .subscribeOn(handlerThread(backgroundThreadHandler)) // Run it on a background thread
+                .observeOn(AndroidSchedulers.mainThread())           // Be notified on the main thread
```
Comment (JakeWharton) [inline sample-app/src/main/java/rx/android/samples/MainActivity.java]: Either static import `mainThread()` or un-static import `handlerThread()`.

**Comment by JakeWharton** [comment]
Awesome. LGTM. Waiting on #184.

**Pull request merged**